contracts/marketplace/nft_listing.ral

```diff
@@ -27,10 +28,18 @@ Contract NFTListing(
         return listingFee
     }
 
+    pub fn requiresRoyalty() -> Bool {
+        return royalty
+    }
+
+    @using(checkExternalCaller = false)
+    pub fn royaltyAmount() -> U256 {
+        return getRoyaltyAmount(tokenId, NFT(tokenId).getCollectionId(), price, royalty)
```

Comment:
We have removed `getCollectionId` from the std, so this should not work now.

Reply:
Yeah, if we use `INFT` then it won't work, since it is not part of the interface anymore. In this case, it is actually a specific `NFT` implementation that still has this method, and this method also does some validity check for the collectionId as well, I have [made a comment](https://github.com/alephium/alephium-nft/blob/nft-royalties/contracts/nft/nft.ral#L17-L22) in the code here. 

WDYT? Or is it cleaner to pass in the collection id?

Reply:
As a generic marketplace, we can only assume that the contract is `INFT`, not `NFT`, right?

Reply:
Great point :+1: